gremlinclient/client.py

```diff
@@ -152,6 +154,9 @@ def parser(f):
                 elif message.status_code == 204:
                     future.set_result(message)
                     self._closed = True
+                elif self._save_errors:
+                    future.set_result(message)
+                    self._closed = True
                 else:
                     future.cancel()
                     raise RuntimeError(
```

Comment:
Here we need to set the exception on the future so the end user can handle. Check out the ``tornado.concurrent`` module and use the ``Future.set_exception`` method. Then you don't need a ``_save_errors`` property.